app/models/concerns/rule_tree.rb

```diff
@@ -4,12 +4,12 @@
 module RuleTree
   extend ActiveSupport::Concern
 
-  RULE_ATTRIBUTES = %i[id ref_id title description rationale identifier severity precedence].freeze
+  RULE_ATTRIBUTES = %i[id ref_id title description rationale identifier references severity precedence].freeze
   RULE_GROUP_ATTRIBUTES = %i[id ref_id title description rationale].freeze
 
   included do
     def rule_tree(graphql = false)
-      rule_groups.includes(:rules).references(:rules).arrange_serializable do |group, children|
+      tree_root.arrange_serializable do |group, children|
         serialize(group, RULE_GROUP_ATTRIBUTES, graphql).merge(
           children: children + group.rules.map do |rule|
             serialize(rule, RULE_ATTRIBUTES, graphql)
```

Comment:
I tested this in the rails console and it looks like there are N+1 queries being generated now. These are the queries being ran over and over: 
```
RuleReferencesContainer Load (0.6ms)  SELECT "rule_references_containers".* FROM "rule_references_containers" WHERE "rule_references_containers"."rule_id" = $1 LIMIT $2  [["rule_id", "ad5e4f9d-ff39-4d83-b8fe-46df1a04c83c"], ["LIMIT", 1]]
  Rule Load (12.8ms)  SELECT "rules".* FROM "rules" WHERE "rules"."rule_group_id" = $1 ORDER BY "rules"."precedence" ASC  [["rule_group_id", "60ffc97e-8972-4152-bd4d-5fcb07988c64"]]
```

Reply:
This should not happen, but I'll check this next week then.